calliope/core/preprocess/model_data.py

```diff
@@ -434,9 +434,16 @@ def group_constraints_to_dataset(model_run):
     # Get a DataFrame with constraint names as index, constraint groups as columns
     constraint_df = (
         pd.DataFrame(model_run['group_constraints'])
-          .drop(['techs', 'locs'], axis=0, errors='ignore')
+          .drop(['techs', 'locs'], axis='index', errors='ignore')
     )
 
+    if 'exists' in constraint_df.index:
+        constraint_df.loc["exists"] = constraint_df.loc["exists"].fillna(True)
```

Comment:
This line creates a Pandas warning:

```
KeyError: '_typ'
Exception ignored in: 'util.is_period_object'
KeyError: ('_typ',)
KeyError: '_typ'
Exception ignored in: 'pandas._libs.lib.is_interval'
KeyError: ('_typ',)
```

When printing `constraint_df` on the command line (e.g. in `pdb`) a similar warning is issued. The reason seems to be the dict within the dataframe. Is it a good idea to store a dict within a Pandas dataframe? Seems brittle.

Reply:
Agreed, is there any reason this is a df at all? Could it not be kept as a dict?